cli/command/stack/kubernetes/deploy.go

```diff
@@ -21,12 +22,27 @@ func RunDeploy(dockerCli *KubeCli, opts options.Deploy) error {
 		return errors.Errorf("Please specify only one compose file (with --compose-file).")
 	}
 
+	// due to env var expansion logic (and the need to conserve the same compose version as original file)
```

Comment:
> the need to conserve the same compose version as original file
Do we need that ?

Reply:
At the beginning, I wanted to just override the original composefile version with the latest supported by the cli (in order to align the "long-formats" generating by marshalling composetypes.Config with the version it is supported in), but after a bit of thinking I found that it was a bad idea: if in a future version of the CLI we introduce a new compose schema version, we will generate composefiles that the server-side can't understand if not updated (and we don't want to break compatibility with Docker EE for example).

Reply:
@simonferquel this is only for `v1beta1` api (compose controller) right ? we know which version we support on that release of the controller, so really, we could, for this case only, decide of a version and *never* update it (until we drop support for `v1beta1` :angel: ). It's a special case anyway so let's make the "fix" the simplest.

Reply:
I don't think keeping an old config structure and implementing conversion from latest to this reference version is simpler than just doing the variable expansion on non-interpreted config files (this PR basically just expose exiting code to the kubernetes subpackage)